// @ddietr/codemirror-themes/dist/theme/tokyo-night-day.cjs
'use strict';

var view = require('@codemirror/view');
var language = require('@codemirror/language');
var highlight = require('@lezer/highlight');

const config = {
    name: 'tokyoNightDay',
    dark: false,
    background: '#e1e2e7',
    foreground: '#6a6f8e',
    selection: '#8591b840',
    cursor: '#3760bf',
    dropdownBackground: '#e1e2e7',
    dropdownBorder: '#6a6f8e',
    activeLine: '#a7aaba22',
    matchingBracket: '#e9e9ec',
    keyword: '#9854f1',
    storage: '#9854f1',
    variable: '#3760bf',
    parameter: '#3760bf',
    function: '#2e7de9',
    string: '#587539',
    constant: '#9854f1',
    type: '#07879d',
    class: '#3760bf',
    number: '#b15c00',
    comment: '#9da3c2',
    heading: '#006a83',
    invalid: '#ff3e64',
    regexp: '#2e5857',
};
const tokyoNightDayTheme = view.EditorView.theme({
    '&': {
        color: config.foreground,
        backgroundColor: config.background,
    },
    '.cm-content': { caretColor: config.cursor },
    '.cm-cursor, .cm-dropCursor': { borderLeftColor: config.cursor },
    '&.cm-focused > .cm-scroller > .cm-selectionLayer .cm-selectionBackground, .cm-selectionBackground, .cm-content ::selection': { backgroundColor: config.selection },
    '.cm-panels': { backgroundColor: config.dropdownBackground, color: config.foreground },
    '.cm-panels.cm-panels-top': { borderBottom: '2px solid black' },
    '.cm-panels.cm-panels-bottom': { borderTop: '2px solid black' },
    '.cm-searchMatch': {
        backgroundColor: config.dropdownBackground,
        outline: `1px solid ${config.dropdownBorder}`
    },
    '.cm-searchMatch.cm-searchMatch-selected': {
        backgroundColor: config.selection
    },
    '.cm-activeLine': { backgroundColor: config.activeLine },
    '.cm-selectionMatch': { backgroundColor: config.selection },
    '&.cm-focused .cm-matchingBracket, &.cm-focused .cm-nonmatchingBracket': {
        backgroundColor: config.matchingBracket,
        outline: 'none'
    },
    '.cm-gutters': {
        backgroundColor: config.background,
        color: config.foreground,
        border: 'none'
    },
    '.cm-activeLineGutter': { backgroundColor: config.background },
    '.cm-foldPlaceholder': {
        backgroundColor: 'transparent',
        border: 'none',
        color: config.foreground
    },
    '.cm-tooltip': {
        border: `1px solid ${config.dropdownBorder}`,
        backgroundColor: config.dropdownBackground,
        color: config.foreground,
    },
    '.cm-tooltip .cm-tooltip-arrow:before': {
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent'
    },
    '.cm-tooltip .cm-tooltip-arrow:after': {
        borderTopColor: config.foreground,
        borderBottomColor: config.foreground,
    },
    '.cm-tooltip-autocomplete': {
        '& > ul > li[aria-selected]': {
            background: config.selection,
            color: config.foreground,
        }
    }
}, { dark: config.dark });
const tokyoNightDayHighlightStyle = language.HighlightStyle.define([
    { tag: highlight.tags.keyword, color: config.keyword },
    { tag: [highlight.tags.name, highlight.tags.deleted, highlight.tags.character, highlight.tags.macroName], color: config.variable },
    { tag: [highlight.tags.propertyName], color: config.function },
    { tag: [highlight.tags.processingInstruction, highlight.tags.string, highlight.tags.inserted, highlight.tags.special(highlight.tags.string)], color: config.string },
    { tag: [highlight.tags.function(highlight.tags.variableName), highlight.tags.labelName], color: config.function },
    { tag: [highlight.tags.color, highlight.tags.constant(highlight.tags.name), highlight.tags.standard(highlight.tags.name)], color: config.constant },
    { tag: [highlight.tags.definition(highlight.tags.name), highlight.tags.separator], color: config.variable },
    { tag: [highlight.tags.className], color: config.class },
    { tag: [highlight.tags.number, highlight.tags.changed, highlight.tags.annotation, highlight.tags.modifier, highlight.tags.self, highlight.tags.namespace], color: config.number },
    { tag: [highlight.tags.typeName], color: config.type, fontStyle: config.type },
    { tag: [highlight.tags.operator, highlight.tags.operatorKeyword], color: config.keyword },
    { tag: [highlight.tags.url, highlight.tags.escape, highlight.tags.regexp, highlight.tags.link], color: config.regexp },
    { tag: [highlight.tags.meta, highlight.tags.comment], color: config.comment },
    { tag: highlight.tags.strong, fontWeight: 'bold' },
    { tag: highlight.tags.emphasis, fontStyle: 'italic' },
    { tag: highlight.tags.link, textDecoration: 'underline' },
    { tag: highlight.tags.heading, fontWeight: 'bold', color: config.heading },
    { tag: [highlight.tags.atom, highlight.tags.bool, highlight.tags.special(highlight.tags.variableName)], color: config.variable },
    { tag: highlight.tags.invalid, color: config.invalid },
    { tag: highlight.tags.strikethrough, textDecoration: 'line-through' },
]);
const tokyoNightDay = [
    tokyoNightDayTheme,
    language.syntaxHighlighting(tokyoNightDayHighlightStyle),
];

exports.config = config;
exports.tokyoNightDay = tokyoNightDay;
exports.tokyoNightDayHighlightStyle = tokyoNightDayHighlightStyle;
exports.tokyoNightDayTheme = tokyoNightDayTheme;
